app/src/main/java/com/google/maps/android/ktx/demo/MainActivity.kt

```diff
@@ -0,0 +1,236 @@
+ *        MAPS_API_KEY="YOUR_API_KEY"
+ */
+class MainActivity : AppCompatActivity() {
+    var mIsRestore: Boolean = false
+
+    @MapsExperimentalFeature
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        mIsRestore = savedInstanceState != null
+        setContentView(R.layout.activity_main)
+        val mapFragment = supportFragmentManager.findFragmentById(R.id.map) as? SupportMapFragment
+        lifecycle.coroutineScope.launchWhenCreated {
+            check(mapFragment != null)
+            val googleMap = mapFragment.awaitMap()
+            if (!mIsRestore) {
+                googleMap.moveCamera(
+                    CameraUpdateFactory.newLatLngZoom(
+                        LatLng(51.403186, -0.126446),
+                        10F
+                    )
+                )
+            }
+            showMapLayers(googleMap)
+        }
+    }
+
+    private fun showMapLayers(map: GoogleMap) {
+        // Shared object managers - used to support multiple layer types on the map simultaneously
+        val markerManager = MarkerManager(map)
+        val groundOverlayManager = GroundOverlayManager(map)
+        val polygonManager = PolygonManager(map)
+        val polylineManager = PolylineManager(map)
+
+        addClusters(map, markerManager)
+        addGeoJson(map, markerManager, polylineManager, polygonManager, groundOverlayManager)
+        addKml(map, markerManager, polylineManager, polygonManager, groundOverlayManager)
+        addMarker(markerManager)
+    }
+
+    private fun addClusters(map: GoogleMap, markerManager: MarkerManager) {
+        val clusterManager = ClusterManager<MyItem>(this, map, markerManager)
+        map.setOnCameraIdleListener(clusterManager)
+
+        try {
+            val items = MyItemReader().read(resources.openRawResource(R.raw.radar_search))
+            clusterManager.addItems(items)
+        } catch (e: JSONException) {
+            Toast.makeText(this, "Problem reading list of markers.", Toast.LENGTH_LONG).show()
+            e.printStackTrace()
+        }
+    }
+
+    private fun addGeoJson(
+        map: GoogleMap,
+        markerManager: MarkerManager,
+        polylineManager: PolylineManager,
+        polygonManager: PolygonManager,
+        groundOverlayManager: GroundOverlayManager
+    ) {
+        // GeoJSON polyline
+        val geoJsonLineLayer = geoJsonLayer(
+            map = map,
+            resourceId = R.raw.south_london_line_geojson,
+            context = this,
+            markerManager = markerManager,
+            polygonManager = polygonManager,
+            polylineManager = polylineManager,
+            groundOverlayManager = groundOverlayManager
+        )
+        // Make the line red
+        val geoJsonLineStringStyle = GeoJsonLineStringStyle()
+        geoJsonLineStringStyle.color = Color.RED
+        for (f in geoJsonLineLayer.features) {
+            f.lineStringStyle = geoJsonLineStringStyle
+        }
```

Comment:
Lines 122-126 can also be written in a functional Kotlin style:

```kotlin
geoJsonLineLayer.features.forEach {
    it.lineStringStyle = GeoJsonLineStringStyle().apply { 
        color = Color.RED
    }
}
```

Reply:
Refactored in https://github.com/googlemaps/android-maps-ktx/pull/29/commits/89b22e8ddd52802b8cc6b102c66d11f3a540c589.